ert_shared/tracker/base.py

```diff
@@ -0,0 +1,178 @@
+import time
+
+from ert_shared.tracker.events import DetailedEvent, EndEvent, GeneralEvent, TickEvent
+from ert_shared.tracker.state import SimulationStateStatus
+from ert_shared.tracker.utils import calculate_progress
+from res.job_queue import JobStatusType
+
+
+class BaseTracker(object):
+    """BaseTracker provides the basis for doing tracking."""
+
+    def __init__(self, model):
+        """Initialize the tracker for a @model. A model can be any
+        BaseRunModel-derived class."""
+        self._model = model
+
+        self._states = []
+        self._custom_states = []
+        self._bootstrap_states()
+        BaseTracker.__checkForUnusedEnums(self._states)
+
+        # keep track of phases in the model using a mapping of phase index
+        # to data is not accessible in the model, but should be.
+        # TODO: rewrite the phases API in BaseRunModel so that this can go away
```

Comment:
Could you make an issue out of this?

Reply:
https://github.com/equinor/ert/issues/556